src/management/Akka.Http.Shim/Dsl/Http.cs

```diff
@@ -0,0 +1,135 @@
+using System.Net;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Annotations;
+using Akka.Configuration;
+using Akka.Http.Dsl.Model;
+using Akka.Http.Dsl.Server;
+using Akka.Http.Dsl.Settings;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Akka.Http.Dsl
+{
+    using HttpRequest = Model.HttpRequest;
+    using HttpResponse = Model.HttpResponse;
+
+    public delegate Task<RouteResult.IRouteResult> Route(RequestContext context);
+    public delegate Route RouteGenerator<T>(T obj);
+    
+    
+    public sealed class HttpExt : IExtension
+    {
+        private readonly Config _config;
+        private readonly ExtendedActorSystem _system;
+
+        public HttpExt(Config config, ExtendedActorSystem system)
+        {
+            _config = config;
+            _system = system;
+        }
+
+        /// <summary>
+        /// Main entry point to create a server binding.
+        /// </summary>
+        /// <param name="hostname">The interface to bind to.</param>
+        /// <param name="port">The port to bind to or `0` if the port should be automatically assigned.</param>
+        public ServerBuilder NewServerAt(string hostname, int port) =>
+            ServerBuilder.Create(hostname, port, _system);
+
+        /// <summary>
+        /// Convenience method which starts a new HTTP server at the given endpoint and uses the given `handler`
+        /// for processing all incoming connections.
+        /// </summary>
+        public async Task<ServerBinding> BindAndHandleAsync(Route route, string hostname, int port, ServerSettings settings)
+        {
+            var localRoute = route;
+            var host = WebHost.CreateDefaultBuilder()
+                .SuppressStatusMessages(true)
+                .ConfigureServices(services =>
+                {
+                    services.AddLogging(builder => builder.AddFilter("Microsoft", LogLevel.Debug));
+                    services.AddRouting();
+                })
+                .Configure(app =>
```

Comment:
Here is a forward compatibility problem with the shim.
In the new ASP.NET, the `Configure(Action<IApplicationBuilder>)` method is burried and is not exposed as an API, the only way to access it is by implementing it inside a `Startup` class, which limits the way we can programatically assemble a server.

Reply:
Turns out that this is not a problem, the missing `Configure` method only applies to the new `IHostBuilder` API, it is still available with the older `IWebHostBuilder` API